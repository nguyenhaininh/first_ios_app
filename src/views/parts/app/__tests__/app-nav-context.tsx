/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import * as React from 'react';
import AppNavContext, { createInitialContextValue } from '../app-nav-context';

import * as renderer from 'react-test-renderer';
import { Router } from 'react-router';
import { createMemoryHistory } from 'history';

it('create initial value', () => {
  const value = createInitialContextValue();
  expect(value).toMatchSnapshot();
});

it('renders correctly', () => {
  const value = createInitialContextValue();
  const tree = renderer.create(
    <AppNavContext.Provider value={value}>
      <AppNavContext.Consumer>
        {(context) => {
          expect(context).toBeTruthy();
          return false;
        }}
      </AppNavContext.Consumer>
    </AppNavContext.Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
