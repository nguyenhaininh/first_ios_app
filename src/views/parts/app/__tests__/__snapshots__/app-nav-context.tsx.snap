// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create initial value 1`] = `
Object {
  "histories": Object {
    "login": Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "jzzz2e",
          "pathname": "/login",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "jzzz2e",
        "pathname": "/login",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    },
    "member": Object {
      "download": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "jzzz2e",
            "pathname": "/member/download",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "jzzz2e",
          "pathname": "/member/download",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "favorite": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "jzzz2e",
            "pathname": "/member/favorite",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "jzzz2e",
          "pathname": "/member/favorite",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "feed": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "jzzz2e",
            "pathname": "/member",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "jzzz2e",
          "pathname": "/member",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "menu": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "jzzz2e",
            "pathname": "/modal/menu",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "jzzz2e",
          "pathname": "/modal/menu",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "ranking": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "jzzz2e",
            "pathname": "/member/ranking",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "jzzz2e",
          "pathname": "/member/ranking",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "search": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "jzzz2e",
            "pathname": "/member/search",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "jzzz2e",
          "pathname": "/member/search",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
  },
}
`;

exports[`renders correctly 1`] = `null`;
